Core/Core/Files/View/FileDetails/FileDetailsViewController.swift

```diff
@@ -308,6 +316,18 @@ extension FileDetailsViewController: URLSessionDownloadDelegate {
         session.finishTasksAndInvalidate()
     }
 
+    private func showFileNoLongerExistsDialog() {
+        let alert = UIAlertController(title: NSLocalizedString("File No Longer Exists", comment: ""),
```

Comment:
We could have this as a universal alert with custom action closure in the ErrorViewController protocol.

Reply:
Unfortunately, it's not that easy. The `FilePicker` class which is an `NSObject` also inherits from that `ErrorViewController` protocol but since it's not a `UIViewController` the default implementation won't be applied there and we would have to copy paste the File No Longer Exists dialog's code there.